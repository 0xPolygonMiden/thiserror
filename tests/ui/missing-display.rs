use thiserror::Error;

#[derive(Error, Debug)]
pub enum Error {
    First,
    Second,
}

fn main() {}
