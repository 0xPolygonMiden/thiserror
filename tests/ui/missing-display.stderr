error[E0277]: `MyError` doesn't implement `std::fmt::Display`
  --> tests/ui/missing-display.rs:4:1
   |
4  | pub enum MyError {
   | ^^^^^^^^^^^^^^^^ `MyError` cannot be formatted with the default formatter
   |
  ::: $RUST/std/src/error.rs
   |
   | pub trait Error: Debug + Display {
   |                          ------- required by this bound in `std::error::Error`
   |
   = help: the trait `std::fmt::Display` is not implemented for `MyError`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
